src/script/interpreter.cpp

```diff
@@ -1587,29 +1689,533 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
+                    if (vchScalar.size() != 32 || !pk.TweakMulVerify(res, uint256(vchScalar)))
+                        return set_error(serror, SCRIPT_ERR_ECMULTVERIFYFAIL);
+
+                    popstack(stack);
+                    popstack(stack);
+                    popstack(stack);
+                }
+                break;
+                case OP_TAPTWEAKVERIFY:
+                {
+                    // OP_TAPTWEAKVERIFY is available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+
+                    valtype& vchInternalKey = stacktop(-1);
+                    valtype& vchTweak = stacktop(-2);
+                    valtype& vchTweakedKey = stacktop(-3);
+
+                    if (vchTweakedKey.size() != CPubKey::COMPRESSED_SIZE || vchInternalKey.size() != 32 || vchTweak.size() != 32)
+                        return set_error(serror, SCRIPT_ERR_PUBKEYTYPE);
+
+                    execdata.m_validation_weight_left -= VALIDATION_WEIGHT_PER_SIGOP_PASSED;
+                    if (execdata.m_validation_weight_left < 0) {
+                        return set_error(serror, SCRIPT_ERR_TAPSCRIPT_VALIDATION_WEIGHT);
+                    }
+                    const XOnlyPubKey tweakedKey{uint256(std::vector<unsigned char>(vchTweakedKey.begin() + 1, vchTweakedKey.begin() + CPubKey::COMPRESSED_SIZE))};
+                    const uint256 tweak(vchTweak);
+                    const XOnlyPubKey internalKey{uint256(vchInternalKey)};
+                    if (!tweakedKey.CheckPayToContract(internalKey, tweak, vchTweakedKey[0] & 1))
+                        return set_error(serror, SCRIPT_ERR_UNKNOWN_ERROR);
+
+                    popstack(stack);
+                    popstack(stack);
+                    popstack(stack);
+                }
+                break;
+
+                case OP_ADD64:
+                case OP_SUB64:
+                case OP_MUL64:
+                case OP_DIV64:
+                case OP_LESSTHAN64:
+                case OP_LESSTHANOREQUAL64:
+                case OP_GREATERTHAN64:
+                case OP_GREATERTHANOREQUAL64:
+                case OP_EQUAL64:
+                case OP_AND64:
+                case OP_OR64:
+                case OP_XOR64:
+                {
+                    // Opcodes only available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+
+                    if (stack.size() < 2)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
+                    valtype& vchb = stacktop(-1);
+                    valtype& vcha = stacktop(-2);
+                    if (vchb.size() != 8 || vcha.size() != 8)
+                        return set_error(serror, SCRIPT_ERR_EXPECTED_8BYTES);
+
+                    int64_t b = ReadLE64(vchb.data());
```

Comment:
ReadLE64 returns an `uint64_t`, but you are assigning it to an `int64_t`.  This conversion is "implementation defined".  I'm not sure how bad that is, but it doesn't sound good.  Does this really not yield a compilier warning?

Reply:
Nope, there is no compiler warning

Reply:
It looks like the functions for 
```
template<typename Stream> inline void Unserialize(Stream& s, int64_t& a) { a = ser_readdata64(s); }
``` in `serialize.h`
already do this implicit cast. So, we are safe in doing this over here?

Reply:
Well since you've written the cast function already we might as well use it.  I'd like to see how this shakes out in Bitcoin Core land, but ya, a direct cast is probably also acceptable.